beacon-chain/blockchain/block_processing.go

```diff
@@ -73,6 +73,17 @@ func (c *ChainService) ReceiveBlock(ctx context.Context, block *ethpb.BeaconBloc
 	if err != nil {
 		return nil, fmt.Errorf("could not hash beacon block")
 	}
+	validatorID, epoch, err := b.ValidateBlockRandaoBeforeStateTransition(beaconState, block.Body)
```

Comment:
don't need `b.ValidateBlockRandaoBeforeStateTransition`, it's already part of state transition, we want to avoid extra validation here for performance reason